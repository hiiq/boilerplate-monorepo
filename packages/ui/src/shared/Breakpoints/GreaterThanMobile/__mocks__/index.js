import React from 'react';

const GreaterThanMobile = (props) => <x-GreaterThanMobile {...props} />;

export { GreaterThanMobile };
