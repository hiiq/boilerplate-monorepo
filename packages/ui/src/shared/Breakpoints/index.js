import { useMediaQuery } from 'react-responsive';

const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};

const Huge = ({ children }) => {
  const isHuge = useMediaQuery({ minWidth: Breakpoint.HUGE });

  return isHuge ? children : null;
};

const Desktop = ({ children }) => {
  const isDesktop = useMediaQuery({ minWidth: Breakpoint.LARGE });

  return isDesktop ? children : null;
};

const Tablet = ({ children }) => {
  const isTablet = useMediaQuery({ minWidth: Breakpoint.MEDIUM });

  return isTablet ? children : null;
};

const Mobile = ({ children }) => {
  const isMobile = useMediaQuery({ maxWidth: Breakpoint.SMALL - 1 });

  return isMobile ? children : null;
};

const GreaterThanMobile = ({ children }) => {
  const isMobile = useMediaQuery({ minWidth: Breakpoint.SMALL });

  return isMobile ? children : null;
};

const Default = ({ children }) => {
  const isNotMobile = useMediaQuery({ minWidth: 0 });

  return isNotMobile ? children : null;
};

export { Default, Desktop, GreaterThanMobile, Huge, Mobile, Tablet };
