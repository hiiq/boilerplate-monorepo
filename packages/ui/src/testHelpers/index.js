import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
