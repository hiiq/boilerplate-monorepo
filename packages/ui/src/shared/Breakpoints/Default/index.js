import { useMediaQuery } from 'shared/useMediaQuery';

const Default = ({ children }) => {
  const isNotMobile = useMediaQuery({ minWidth: 0 });

  return isNotMobile ? children : null;
};

export { Default };
