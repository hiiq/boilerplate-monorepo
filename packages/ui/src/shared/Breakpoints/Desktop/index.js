import { Breakpoint } from '@boilerplate-monorepo/ui-common';
import { useMediaQuery } from 'shared/useMediaQuery';

const Desktop = ({ children }) => {
  const isDesktop = useMediaQuery({ minWidth: Breakpoint.LARGE });

  return isDesktop ? children : null;
};

export { Desktop };
