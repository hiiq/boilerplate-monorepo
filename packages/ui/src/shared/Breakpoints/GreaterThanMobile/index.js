import { Breakpoint } from '@boilerplate-monorepo/ui-common';
import { useMediaQuery } from 'shared/useMediaQuery';

const GreaterThanMobile = ({ children }) => {
  const isMobile = useMediaQuery({ minWidth: Breakpoint.SMALL });

  return isMobile ? children : null;
};

export { GreaterThanMobile };
