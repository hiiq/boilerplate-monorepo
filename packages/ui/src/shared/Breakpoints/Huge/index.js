import { Breakpoint } from '@boilerplate-monorepo/ui-common';
import { useMediaQuery } from 'shared/useMediaQuery';

const Huge = ({ children }) => {
  const isHuge = useMediaQuery({ minWidth: Breakpoint.HUGE });

  return isHuge ? children : null;
};

export { Huge };
