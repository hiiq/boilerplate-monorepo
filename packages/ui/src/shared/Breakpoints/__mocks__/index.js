import { Desktop } from '../Desktop';
import { GreaterThanMobile } from '../GreaterThanMobile';
import { Huge } from '../Huge';
import { Mobile } from '../Mobile';
import { Tablet } from '../Tablet';

const Breakpoints = {
  Desktop,
  GreaterThanMobile,
  Huge,
  Mobile,
  Tablet,
};

export { Breakpoints };
