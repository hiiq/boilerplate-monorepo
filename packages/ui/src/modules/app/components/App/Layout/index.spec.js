import React from 'react';
import * as GreaterThanMobileBreakpoint from 'shared/Breakpoints/GreaterThanMobile';
import { render } from 'testHelpers';
import { Layout } from '.';

jest.mock('shared/useAuth');
jest.mock('./FavIcon', () => ({
  FavIcon: (props) => <x-FavIcon {...props} />,
}));
jest.mock('./Footer', () => ({
  Footer: (props) => <x-Footer {...props} />,
}));
jest.mock('./Header', () => ({
  Header: (props) => <x-Header {...props} />,
}));
jest.mock('./Main', () => ({
  Main: (props) => <x-Main {...props} />,
}));
jest.mock('./Meta', () => ({
  Meta: (props) => <x-Meta {...props} />,
}));
jest.mock('./Navigation', () => ({
  Navigation: (props) => <x-Navigation {...props} />,
}));

describe('<Layout/>', () => {
  const defaultProps = {};

  const renderComponent = (overrides) =>
    render(<Layout {...defaultProps} {...overrides} />);

  describe('when on mobile', () => {
    test('renders properly', () => {
      td.replace(GreaterThanMobileBreakpoint, 'GreaterThanMobile', () => null);

      const { container } = renderComponent();

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when on larger than mobile', () => {
    test('renders properly', () => {
      td.replace(GreaterThanMobileBreakpoint, 'GreaterThanMobile', (props) => (
        <x-GreaterThanMobile {...props} />
      ));

      const { container } = renderComponent();

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
