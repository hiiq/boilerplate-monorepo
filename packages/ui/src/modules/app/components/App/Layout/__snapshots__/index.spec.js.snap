// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> when on larger than mobile renders properly 1`] = `
<div
  class="Layout__Container-e3nv7w-0 jLGuRX"
>
  <x-meta />
  <x-favicon />
  <x-header />
  <x-greaterthanmobile>
    <x-navigation />
  </x-greaterthanmobile>
  <x-main />
  <x-footer />
</div>
`;

exports[`<Layout/> when on mobile renders properly 1`] = `
<div
  class="Layout__Container-e3nv7w-0 jLGuRX"
>
  <x-meta />
  <x-favicon />
  <x-header />
  <x-main />
  <x-footer />
</div>
`;
